### PR TITLE
Adding command capabilities

### PR DESCRIPTION
**Pull request by PythooonUser**
## Summary
More work regarding capability support with commands, see #16.

## Changes
- Adding more event types to the TwitchChatEvent class
- Adding stub for wrapper methods regarding those new commands

### CODE REVIEW

File: twitchobserver/twitchobserver.py [hunk 1]
```diff
@@ -54,8 +70,13 @@ def __init__(self, channel=None, command=None, message=''):
 
     def dumps(self):
         message = getattr(self, 'message', '')
+
         if message:
-            message = ' :' + message
+            if self.type == "TWITCHCHATHOSTTARGET":
```
Comment (JoshuaSkelly) [inline twitchobserver/twitchobserver.py]: I don't think the user can send this event. Can you confirm?

File: twitchobserver/twitchobserver.py [hunk 2]
```diff
@@ -162,6 +183,11 @@ def leave_channel(self, channel):
         
         self._send_events(TwitchChatEvent(channel, 'PART'))
 
+    def clear_chat(self, channel, user=None):
```
Comment (JoshuaSkelly) [inline twitchobserver/twitchobserver.py]: Can the user actually send this message?

File: twitchobserver/twitchobserver.py (hunk 1 above)
Comment (PythooonUser) [inline twitchobserver/twitchobserver.py]: When sending  CLEARCHAT I get a 421 error, i.e. unkown command.

**Comment by PythooonUser** [comment]
Also the events 001, 002, 003 etc. are now processed as events. I will look into that.

**Comment by JoshuaSkelly** [comment]
Don't worry about those event. It is by design and from the changes I made. If you want to change that behavior, open an issue and we can talk through the options.

Keep your work in this PR constrained to the task. 😄

**Comment by PythooonUser** [comment]
Yes, the API also confirms it: those additional events are inbound only.

**Comment by JoshuaSkelly** [comment]
Can you add WHISPER to _process_server_messages()?

**Comment by JoshuaSkelly** [comment]
Also I think we can support the outbound messages though PRIVMSG?

I tested send_message('/w user message', 'channel') and it did successfully send a whisper.

**Comment by PythooonUser** [comment]
What about the regex fro WHISPER?

**Comment by PythooonUser** [comment]
Do you want to have it its own wrapper or leave it with send_message()?

**Comment by JoshuaSkelly** [comment]
The regex below should work.

```python
# WHISPER parameters. Groups: (recipient, message)
_whisper_params_re = re.compile('(\w+)\s+:([\s\S]*)')
```

**Comment by JoshuaSkelly** [comment]
I was thinking a nice helper method like:
```python
def send_whisper(self, nickname, message):
    pass
```

Which could wrap the PRIVMSG call, or however best you think to implement it.

**Comment by PythooonUser** [comment]
If found https://help.twitch.tv/customer/portal/articles/659095-chat-moderation-commands and working my way trough it.

**Comment by JoshuaSkelly** [comment]
Hmm. Maybe keep your changes scoped to the commands specific to the user story?

**Comment by PythooonUser** [comment]
Only for those we have events for? Like /ban and /unban which is CLEARCHAT.

**Comment by JoshuaSkelly** [comment]
Only for the events listed in the user story #16 

I have questions about CLEARCHAT. The documentation makes it seem like /ban, except in one place where it implies it works like /clear

**Comment by PythooonUser** [comment]
Yes. CLEARCHAT will be called by /ban and /clear. /clear wipes out the chat and /ban bans a user from chat. /unban does not result in an event call unfortunately.

**Comment by PythooonUser** [comment]
So there are two methods for banning users? Calling CLEARCHAT and /ban? There is no equivalent to /unban?

**Comment by JoshuaSkelly** [comment]
I don't think we can call CLEARCHAT? I think we issue a /ban and the server sends back CLEARCHAT message?

**Comment by PythooonUser** [comment]
Ehm, correct. We already sorted that out. Those additional commands are inbound only. It's getting late... :)

**Comment by JoshuaSkelly** [comment]
No worries man. Don't kill yourself, get some rest. 😴

File: twitchobserver/twitchobserver.py [hunk 3]
```diff
@@ -162,6 +176,31 @@ def leave_channel(self, channel):
         
         self._send_events(TwitchChatEvent(channel, 'PART'))
 
+    def whisper(self, user, message):
```
Comment (JoshuaSkelly) [inline twitchobserver/twitchobserver.py]: Should this be ```send_whisper```?

File: twitchobserver/twitchobserver.py [hunk 4]
```diff
@@ -202,6 +242,7 @@ def inbound_worker():
                 try:
                     with self._socket_lock:
                         response = self._socket.recv(1024).decode('utf-8')
+                        #print("RESPONSE\n{}".format(response))
```
Comment (JoshuaSkelly) [inline twitchobserver/twitchobserver.py]: Remove this.

File: twitchobserver/twitchobserver.py [hunk 5]
```diff
@@ -304,14 +345,19 @@ def _process_server_messages(self, response):
...
 
-                elif cmd == 'PRIVMSG':
+                elif cmd in ('PRIVMSG', 'HOSTTARGET', 'NOTICE', 'USERNOTICE'):
                     channel, message = _privmsg_params_re.match(args).groups()
                     event.channel = channel
                     event.message = message
 
+                elif cmd == "WHISPER":
+                    recipient, message = _whisper_params_re.match(args).groups()
+                    event.recipient = recipient
```
Comment (JoshuaSkelly) [inline twitchobserver/twitchobserver.py]: I don't think we need this attribute. It appears to always be the name of the observer. If this isn't the case let me know.

**Comment by JoshuaSkelly** [comment]
Looks good to me! Feel free to commit at your earliest convenience. 👍

**Pull request merged**